keys/signing_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/coinbase/rosetta-sdk-go/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func hashAndHexEncode(message string) string {
+	messageHashBytes := common.BytesToHash([]byte(message)).Bytes()
+	return hex.EncodeToString(messageHashBytes)
+}
+
+func TestSignEd25519(t *testing.T) {
+	curve := types.Edwards25519
+	keypair, err := GenerateKeypair(curve)
+	assert.NoError(t, err)
+
+	signatureType := types.Ed25519
+	payload := &types.SigningPayload{
+		Address:       "test",
+		HexBytes:      "68656C6C6F0D0A",
+		SignatureType: signatureType,
+	}
+
+	signature, err := SignPayload(payload, keypair)
```

Comment:
Do you think it's worth making any assertions on the signature length since that's one of the things that changese between sig types?